wagtail/actions/unpublish_page.py

```diff
@@ -96,7 +96,11 @@ def execute(self, skip_permission_checks=False):
 
             user_perms = UserPagePermissionsProxy(self.user)
             for live_descendant_page in (
-                self.page.get_descendants().live().defer_streamfields().specific()
```

Comment:
> Since this signal is mostly used for cache invalidation, which doesn't usually require a specific page
That's probably what the original intention was, but as someone who has used signals for all sorts of random client requirements in the past, I wouldn't hold too strongly onto this assumption.

Reply:
You do raise an interesting point around consistency in the use of specific pages in signals. One avenue for improving things here might be to always use `Page.specific_class` as the `sender` value when emitting - even if the object itself remains a generic `Page`. I think it would be quite reasonable to make devs do `obj.specific` if they definitely want the specific version.

Reply:
If we drop `.specific()`, we'll want an upgrade consideration note as I would not be surprised if people are using the specific version for whatever custom needs they have